Updates to order management (#113)
vippsas/vipps-order-management-api: vipps-order-management-api.md

<!-- END_COMMENT -->

> The Order Management API allows merchants to enrich Vipps MobilePay transactions.

Reviewer: Maybe just skip the company name?
```suggestion
The Order Management API allows merchants to enrich payments.
```

The information given in this API will be shown to the customer in the order
history in their app. Order Management operates with two concepts:
[Categories](#categories) (with images) and [Receipts](#receipts).
These concepts may be used separately or combined, and this guide will explain how to add them.

This information is shown to the customer in the app in their order history and immediately after in-store purchases.

Order Management enables you to communicate with your customers through

> the payment receipts in the Vipps or MobilePay app. The purpose of doing this is to give

Reviewer: Maybe just skip the company name?
```suggestion
the payment receipts in the app. The purpose of doing this is to give
```

your customers more convenience, better overview and a more compelling shopping
experience when they use Vipps MobilePay to pay for your products and services.
Order management also enables you to draw customers back to your website
or app from links on the receipt view.

This functionality is available for
[recurring](https://developer.vippsmobilepay.com/docs/APIs/recurring-api)
and
[direct payments](https://developer.vippsmobilepay.com/docs/APIs/epayment-api),
but not for
[pass-through payments](https://developer.vippsmobilepay.com/docs/APIs/psp-api).

API version: 2.3.0.

## Before you begin

Sign up your organization and retrieve your API credentials for the
[test environment](https://developer.vippsmobilepay.com/docs/vipps-developers/test-environment)
from
[portal.vipps.no](https://portal.vipps.no).

### Authentication

> All API calls are authenticated with an access token and an API subscription key.

Reviewer: ```suggestion
All API requests are authenticated with an access token and an API subscription key.
```

See
[Get an access token](https://developer.vippsmobilepay.com/docs/APIs/access-token-api#get-an-access-token), for details.

### HTTP headers

We recommend using the standard HTTP headers for all requests

See [HTTP headers](https://developer.vippsmobilepay.com/docs/vipps-developers/common-topics/http-headers)
in the Getting started guide, for details.

### Reference and PaymentType

> Use order management to add a *Receipt* or *Category* to a transaction made with the ePayment, eCom, or Recurring API. The receipt is connected to a `OrderId` (called `reference` in the new ePayment API). The `OrderId` is what **you** use when either initiating an eCom payment or creating a recurring charge.

Reviewer: I hope its lowercase o:

```suggestion
Use order management to add a *Receipt* or *Category* to a transaction made with the ePayment, eCom, or Recurring API. The receipt is connected to a `orderId` (called `reference` in the new ePayment API). The `orderId` is what **you** use when either initiating an eCom payment or creating a recurring charge.
```

> Use order management to add a *Receipt* or *Category* to a transaction made with the ePayment, eCom, or Recurring API. The receipt is connected to a `OrderId` (called `reference` in the new ePayment API). The `OrderId` is what **you** use when either initiating an eCom payment or creating a recurring charge.

Reviewer: ```suggestion
Use order management to add a *Receipt* or *Category* to a transaction made with the ePayment, eCom, or Recurring API. The receipt is connected to a `orderId` (called `reference` in the new ePayment API). The `orderId` is what **you** use when either initiating an eCom payment or creating a recurring charge.
```

Author: I don't know if we can make this change because they have used `orderId` everywhere, also in the yaml file. I think it will be more confusing to the user to use `reference` in that case.

Author: That needs to be done in a different PR.

The Order Management API does **no validation if the order exists**. This means that the order management enrichment may be added before or after the payment is actually created. The preferred way is to add the Order Management Data simultaneously as you initiate the payment.

> As the same `OrderId` can be used for both a Recurring charge and an eCom payment, you need to supply which Vipps product is being used by setting the appropriate `paymentType` - which is either `ecom` or `recurring`.

Reviewer: Lowercase, I hope: ```suggestion
As the same `orderId` can be used for both a Recurring charge and an eCom payment, you need to supply which Vipps product is being used by setting the appropriate `paymentType` - which is either `ecom` or `recurring`.
```
